### PR TITLE
Model paginator optimization

### PR DESCRIPTION
**Pull request by ekmst**
Hello!

* Type: new feature | optimization | performance
* Link to issue:

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:

Rewrote `Paginator\Adapter\Model` using offset.
Since the previous implementation received all the records and paged it at the application level. Below is an example of queries:

```php
use Phalcon\Paginator\Adapter\Model;

$paginator = new Model (
    [
        'data' => Test :: find (
            [
                'columns' => 'id, column, column1'
            ]
        ),
        'limit' => 3,
        'page' => 4,
    ]
);

```

```sql
SELECT `test`.`id`,` test`.`column`, `test`.`column1` FROM` test`
```

And here is an example request using offset:

```php
use Phalcon\Paginator\Adapter\Model;

$paginator = new Model (
    [
        'model' => Test :: class,
        'parameters' => [
            'columns' => 'id, column, column1',
            'limit'  => 12,  // will be ignored
        ],
        'limit' => 3,
        'page' => 4,
    ]
);

```

```sql
SELECT `test`.`id`,` test`.`column`, `test`.`column1` FROM` test` LIMIT: APL0 OFFSET: APL1

-- [APL0] => 3
-- [APL1] => 9
```

But I had to change the `Model` adapter settings. Which breaks backward compatibility with the previous version

What do you think of it?

Thanks

### CODE REVIEW

**Comment by niden** [comment]
@ekmst if you are going to break backwards compatibility now is the time to do it before v4 is out.

Go ahead with this. We will also need to update the docs and note the BC change in the upgrade document

File: phalcon/Paginator/Adapter/Model.zep [hunk 1]
```diff
@@ -41,64 +81,39 @@ class Model extends AbstractAdapter
...
+        if rowcount > 0 {
+            let parameters["limit"]  = limit,
+                parameters["offset"] = limit * (pageNumber - 1);
 
-                items->next();
-            }
+            let items = <ResultsetInterface> call_user_func(
+                [modelClass, "find"],
+                parameters
+            );
```
Comment (niden) [inline phalcon/Paginator/Adapter/Model.zep]: @sergeyklay isnt' 

```
{modelClass}::find(parameters);
```

faster than `call_user_func` ?

Comment (sergeyklay) [reply phalcon/Paginator/Adapter/Model.zep]: No, in that case call_user_func will be slover

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
Thank you @ekmst

**Pull request merged**